src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result = items.reduce((groupedByValues: GroupsMap<T>, item: T) => {
+    const groupKey = String(item[key]);
+
+    if (!groupedByValues[groupKey]) {
+      groupedByValues[groupKey] = [];
+    }
+    groupedByValues[groupKey].push(item);
+
+    return groupedByValues;
+  }, {});
```

Comment:
The reduce callback function can be simplified by using optional chaining and the nullish coalescing operator. This will make the code more concise and easier to read.